src/libraries/System.Net.Sockets/src/System/Net/Sockets/UDPClient.cs

```diff
@@ -639,6 +672,27 @@ async Task<UdpReceiveResult> WaitAndWrap(Task<SocketReceiveFromResult> task)
             }
         }
 
+        public ValueTask<UdpReceiveResult> ReceiveAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+
+            return WaitAndWrap(_clientSocket.ReceiveFromAsync(
+                _buffer,
+                SocketFlags.None,
+                _family == AddressFamily.InterNetwork ? IPEndPointStatics.Any : IPEndPointStatics.IPv6Any, cancellationToken));
+
+            async ValueTask<UdpReceiveResult> WaitAndWrap(ValueTask<SocketReceiveFromResult> task)
```

Comment:
Ok, I realized it's most likely there to keep `ObjectDisposedExceptions` synchronous, nevermind the question if that's the case.

Reply:
@antonfirsov Just wonder, even if there is no WaitAndWrap() and just await in parent method, the **ObjectDisposedExceptions** can still be thrown synchronous because the throw is before first await. Please correct me if wrong, thanks.

Reply:
Never mind, I got reason for that.